subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan_aux.c

```diff
@@ -785,3 +796,239 @@ bool isr_scan_connect_rsp_check(struct pdu_adv *pdu_tx, struct pdu_adv *pdu_rx)
 		       pdu_tx->connect_ind.init_addr, BDADDR_SIZE) == 0);
 }
 #endif /* CONFIG_BT_CENTRAL */
+
+struct node_rx_pdu *lll_scan_aux_setup(struct lll_scan *lll,
+				       struct pdu_adv *pdu)
+{
+	struct pdu_adv_com_ext_adv *pri_com_hdr;
+	struct pdu_adv_ext_hdr *pri_hdr;
+	struct pdu_adv_aux_ptr *aux_ptr;
+	struct node_rx_pdu *node_rx;
+	uint32_t window_widening_us;
+	uint32_t window_size_us;
+	struct node_rx_ftr *ftr;
+	uint32_t aux_offset_us;
+	uint32_t aux_start_us;
+	uint32_t radio_end_us;
+	uint8_t *pri_dptr;
+	uint8_t phy;
+
+	/* FIXME: this always uses pri-to-aux phy, need to setup correctly */
+
+	LL_ASSERT(pdu->type == PDU_ADV_TYPE_EXT_IND);
+
+	/* Get reference to extended header */
+	pri_com_hdr = (void *)&pdu->adv_ext_ind;
+	pri_hdr = (void *)pri_com_hdr->ext_hdr_adv_data;
+	pri_dptr = pri_hdr->data;
+
+	/* traverse through adv_addr, if present */
+	if (pri_hdr->adv_addr) {
+		pri_dptr += BDADDR_SIZE;
+	}
+
+	/* traverse through adi, if present */
+	if (pri_hdr->adi) {
+		pri_dptr += sizeof(struct pdu_adv_adi);
+	}
+
+	/* No need to scan further if no aux_ptr filled */
+	aux_ptr = (void *)pri_dptr;
+	if (unlikely(!pri_hdr->aux_ptr || !aux_ptr->offs)) {
+		return NULL;
+	}
+
+	/* Determine the window size */
+	if (aux_ptr->offs_units) {
+		window_size_us = 300U;
+	} else {
+		window_size_us = 30U;
+	}
+
+	/* Calculate the aux offset from start of the scan window */
+	aux_offset_us = (uint32_t)aux_ptr->offs * window_size_us;
+	radio_end_us = radio_tmr_end_get() -
+		       radio_rx_chain_delay_get(lll->phy, 1);
+
+	/* Calculate the window widening that needs to be deducted */
+	if (aux_ptr->ca) {
+		window_widening_us = aux_offset_us / 2000U;
+	} else {
+		window_widening_us = aux_offset_us / 20000U;
+	}
+
+	phy = BIT(aux_ptr->phy);
+
+	aux_start_us = radio_end_us;
+	aux_start_us -= PKT_AC_US(pdu->len, 0, lll->phy);
+	aux_start_us += aux_offset_us;
+	aux_start_us -= lll_radio_rx_ready_delay_get(phy, 1);
+	aux_start_us -= window_widening_us;
+	aux_start_us -= EVENT_JITTER_US;
+
+	/* Sufficient offset to ULL schedule the auxiliary PDU scan? */
+	if ((aux_start_us < radio_end_us) ||
+	    ((aux_start_us - radio_end_us) >
+	     (HAL_TICKER_TICKS_TO_US(HAL_TICKER_CNTR_CMP_OFFSET_MIN << 1) +
+	      EVENT_OVERHEAD_START_US))) {
+//		printk("LLL SCAN AUX SKIP %u\n", (aux_offset_us - radio_end_us));
+		return NULL;
+	}
+
+	/* LLL schedule the auxiliary PDU scan */
+//	printk("LLL SCAN AUX %u\n", (aux_offset_us - radio_end_us));
+
+	node_rx = ull_pdu_rx_alloc_peek(1);
+	LL_ASSERT(node_rx);
+
+	/* Store the lll context, aux_ptr and start of PDU in footer */
+	ftr = &(node_rx->hdr.rx_ftr);
+	ftr->param = lll;
+	ftr->extra = aux_ptr;
+	ftr->radio_end_us = radio_tmr_end_get() -
+			    PKT_AC_US(pdu->len, 0, lll->phy);
+
+	radio_isr_set(isr_scan_aux_setup, node_rx);
+	radio_disable();
```

Comment:
Yes, because we need different radio disable handling (we'll schedule aux scan from there) than in standard case. The return value will tell caller to skip standard radio disable routine.